plugin/adsk/plugin/exportTranslator.cpp

```diff
@@ -0,0 +1,211 @@
+//
+// Copyright 2016 Pixar
+// Copyright 2019 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "exportTranslator.h"
+
+#include <mayaUsd/fileio/jobs/jobArgs.h>
+#include <mayaUsd/fileio/shading/shadingModeRegistry.h>
+#include <mayaUsd/fileio/jobs/writeJob.h>
+#include <mayaUsd/fileio/utils/writeUtil.h>
+
+#include <maya/MFileObject.h>
+#include <maya/MGlobal.h>
+#include <maya/MSelectionList.h>
+#include <maya/MString.h>
+
+#include <string>
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+MAYAUSD_NS_DEF {
+
+const MString UsdMayaExportTranslator::translatorName("USD Export");
+
+void* UsdMayaExportTranslator::creator() {
+    return new UsdMayaExportTranslator();
+}
+
+UsdMayaExportTranslator::UsdMayaExportTranslator() :
+        MPxFileTranslator() {
+}
+
+UsdMayaExportTranslator::~UsdMayaExportTranslator() {
+}
+
+MStatus
+UsdMayaExportTranslator::writer(const MFileObject &file, 
+                 const MString &optionsString,
+                 MPxFileTranslator::FileAccessMode mode ) {
+
+    std::string fileName(file.fullName().asChar(), file.fullName().length());
+    VtDictionary userArgs;
+    bool exportAnimation = false;
+    GfInterval timeInterval(1.0, 1.0);
+    double frameStride = 1.0;
+    bool append=false;
+    
+    MStringArray filteredTypes;
+    // Get the options 
+    if ( optionsString.length() > 0 ) {
```

Comment:
why the spaces between the brackets and the content? I don't see this style in the rest of this file

Reply:
I think it was only because I did most of the work in one editor that was automatically enforcing a style, and the flipped to another for a bit.

Reply:
Actually...it was like this because that's also how it is in the Pixar plugin code I copied it from.